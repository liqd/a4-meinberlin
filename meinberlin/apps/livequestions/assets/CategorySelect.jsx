import django from 'django'
import React from 'react'

export default class SelectCategory extends React.Component {
  componentDidMount () {
    const select = $('#categorySelect')
    select.on('change', this.props.onSelect)
  }

  render () {
    return (
      <div>
        {Object.keys(this.props.category_dict).length > 0 &&
          <div className="live_questions__select u-spacer-bottom">
            <label htmlFor="categorySelect">{django.gettext('Affiliation')}*</label>
            <div className="form-hint">
              {django.gettext('Answered questions will be displayed in the statistics according to the chosen affiliation.')}
            </div>
            <select
              name="categorySelect"
              id="categorySelect"
              className="js-select2"
              required="required"
              data-minimum-results-for-search="Infinity"
            >
              <option value="">--------</option>
              {Object.keys(this.props.category_dict).map((categoryPk, index) => {
                return <option key={index} value={categoryPk}>{this.props.category_dict[categoryPk]}</option>
              })}
            </select>
          </div>}
      </div>
    )
  }
}
