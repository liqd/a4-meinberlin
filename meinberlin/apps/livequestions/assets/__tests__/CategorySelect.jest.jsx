import React from 'react'
import { render } from '@testing-library/react'

import { CategorySelect } from '../CategorySelect'

test('CategorySelect renders correctly', () => {
  const { asFragment } = render(
    <CategorySelect
      name="Category"
      category_dict={{ 123: 'category1', 124: 'category2' }}
    />)
  expect(asFragment()).toMatchSnapshot()
})
