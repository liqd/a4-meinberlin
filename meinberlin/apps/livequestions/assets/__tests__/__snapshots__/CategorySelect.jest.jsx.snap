// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategorySelect renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="live_questions__select u-spacer-bottom"
    >
      <label
        for="categorySelect"
      >
        Affiliation*
      </label>
      <div
        class="form-hint"
      >
        Answered questions will be displayed in the statistics according to the chosen affiliation.
      </div>
      <select
        class="form-select"
        data-minimum-results-for-search="Infinity"
        id="categorySelect"
        name="categorySelect"
        required=""
      >
        <option
          value=""
        >
          --------
        </option>
        <option
          value="123"
        >
          category1
        </option>
        <option
          value="124"
        >
          category2
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;
